File: 1894-merge-strings-alternately/merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        int n = word1.length();
        int m = word2.length();
        int i = 0;
        int size = min(n, m);
        string ans = "";
        while(i < size)
        {
            ans+= word1[i];
            ans+= word2[i];
            i+=1;
        }
        while(i < n)
            {
                ans+= word1[i];
                i+=1;
            }
        while(i < m)
        {
            ans+= word2[i];
            i+=1;
        }
        return ans;
    }
};

File: 1894-merge-strings-alternately/merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        res = ""
        l, r = 0, 0 
        n, m = len(word1), len(word2)
        while l < n and r < m:
            res += word1[l]
            res += word2[r]
            l+=1
            r+=1 
        if l < n:
            while l < n:
                res += word1[l]
                l+=1
        if r < m:
            while r < m:
                res += word2[r]
                r += 1
        return res